x-pack/modules/azure/lib/filters/azure_event.rb

```diff
@@ -0,0 +1,246 @@
+        category = event.get("[@metadata][category]")
+        if category.nil?
+          category = find_category(event)
+        end
+        event.set("[azure][category]", category) if category
+        group = @activity_log_categories.include?(category) ? ACTIVITY_LOG_GROUP : @sql_diagnostics_categories.include?(category) ? SQL_DIAGNOSTICS_GROUP : nil
+        event.set("[azure][group]", group) if group
+
+        !category.nil?
+      end
+
+      ##
+      # Based on the group and category, explode the custom properties bag
+      def explode_properties(group, category, event)
+        if group == ACTIVITY_LOG_GROUP && category == "ServiceHealth"
+          add_json_event_properties(event)
+        end
+      end
+
+      # Some events such as Service Health can have a Json inside Json as [properties][eventProperties]. This will explode the inner json and replace it with indivisual fields on the event.
+      def add_json_event_properties(event)
+        inner_json = event.get("[properties][eventProperties]")
+        inner_json_as_hash = json_to_hash(inner_json)
+        if inner_json_as_hash
+          inner_json_as_hash.each do |k, v|
+            event.set("[properties][#{k}]", v)
+          end
+          event.remove("[properties][eventProperties]")
+        end
+      end
+
+      # private methods should not change the state of the event
+      private
+
+      CONST_FIELD_NAMES = {
+          "SUBSCRIPTIONS" => "subscription",
+          "RESOURCEGROUPS" =>"resource_group",
+          "PROVIDERS" => "provider"
+      }.freeze
+
+      ##
+      # In general the format of the resourceID for all types is:
+      # /SUBSCRIPTIONS/<subscription>/RESOURCEGROUPS/<resource_group>/PROVIDERS/<provider>/<resource_type1>/<resource_name1>/<resource_typeN/<resource_nameN>
+      # However, it can be something as short as
+      # /SUBSCRIPTIONS/<subscription>
+      # It was also observed to be:
+      # /SUBSCRIPTIONS/<subscription>/PROVIDERS/<provider>
+      # Or as long as needed where the resource_type/resource_name pairings can continue N times.
+      # The display values for repeating resource_type/resource_name pairings should be:
+      # resource_type = resource_type1/resource_type2/resource_typeN (concat the types separated by a "/")
+      # resource_name = resource_nameN (don't concat the names, only use the last)
+      # returns a hash of the key/value pairings with VALUE always UPPERCASE, returns nil if there are not at least 2 parts
+      ##
+      def explode_common_resources(resource_id)
+        logger.debug("parsing resourceId={}", resource_id)
+        grouped_parts = {}
+        parts = resource_id.upcase.split("/")
+        return nil unless parts.size >= 3
+        return nil unless parts.shift == "" # split returns an empty first result when the string starts with the split value
+
+        index = 1
+        const_count = 0
+        const_count += 1 if resource_id.upcase.include?("/SUBSCRIPTIONS/")
```

Comment:
we upcase `resource_id` 4 times in this method, maybe we could just start this method with `resource_id = resource_id.upcase`